src/profile-logic/merge-compare.js

```diff
@@ -1098,3 +1103,55 @@ function mergeMarkers(
 
   return { markerTable: newMarkerTable, translationMaps };
 }
+
+/**
+ * Merge screenshot markers from different threads. And update the target threads string table while doing it.
+ */
+function mergeScreenshotMarkers(
+  threads: Thread[],
+  targetThread: Thread
+): {
+  markerTable: RawMarkerTable,
+  translationMaps: TranslationMapForMarkers[],
+} {
+  const targetMarkerTable = { ...targetThread.markers };
+  const translationMaps = [];
+
+  threads.forEach(thread => {
+    if (thread.name === 'Compositor') {
+      const translationMap = new Map();
+      const { markers, stringTable } = thread;
+
+      for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+        // We need to move the url string to the new string table if doesn't exist.
+        if (
+          markers.data[markerIndex] !== null &&
+          markers.data[markerIndex].type === 'CompositorScreenshot'
+        ) {
+          const urlIndex = markers.data[markerIndex].url;
+          const newUrl = urlIndex >= 0 ? stringTable.getString(urlIndex) : null;
+
+          // Move compositorScreenshot marker data to the new marker table.
+          targetMarkerTable.data.push(markers.data[markerIndex]);
+          targetMarkerTable.data.url =
+            newUrl === null
+              ? -1
+              : targetThread.stringTable.indexForString(newUrl);
+
+          targetMarkerTable.name.push(markers.name[markerIndex]);
+          targetMarkerTable.startTime.push(markers.startTime[markerIndex]);
+          targetMarkerTable.endTime.push(markers.endTime[markerIndex]);
+          targetMarkerTable.phase.push(markers.phase[markerIndex]);
+          targetMarkerTable.category.push(markers.category[markerIndex]);
+
+          // Set the translation map and increase the table length.
+          translationMap.set(markerIndex, targetMarkerTable.length);
+          targetMarkerTable.length++;
+        }
+      }
+      translationMaps.push(translationMap);
```

Comment:
I couldn't understand much about `translationMaps` here. So left it as it is in the `mergeMarkers()`.